### PR TITLE
Chisel tester support

### PR DESCRIPTION
**Pull request by chick**
…subclass which allows tester to have clean up and other construction code executed after a user code executed during constructor of that subclass.
Created a spec file TesterDriverSpec that shows finish method being called

### CODE REVIEW

File: src/test/scala/chiselTests/TesterDriverSpec.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import Chisel.testers.BasicTester
+
+/**
+  * Created by chick on 2/8/16.
+  */
+class FinishTester extends BasicTester {
+  var finish_was_run = false
+
+  override def finish(): Unit = {
+    finish_was_run = true
```
Comment (chick) [inline src/test/scala/chiselTests/TesterDriverSpec.scala]: I'd like to do something like this but it is difficult given the overall test harness now contained in chisel3.  If you have some idea of how to show this, I can try to implement

**Comment by ducky64** [comment]
retest this please

**Comment by chick** [comment]
The problem appears to be due to changes in firrtl.  I was running an older version, specifically:
```
commit f3c4c604549db6bbe824c29649bb05bba7470d8a
Author: jackkoenig <jack.koenig3@eecs.berkeley.edu>
Date:   Fri Jan 29 11:08:14 2016 -0800

    Add project/build.properties with requirement of sbt 0.13.6, this version is required for sbt assembly to work
```
When I pulled the latest changes to take me to:
```
commit 404b5a18483e86088e29b333cc61d6628ae50d15
Author: azidar <azidar@gmail.com>
Date:   Tue Feb 9 19:15:25 2016 -0800

    Updated Readme
```
When I run
```
sbt
> test-only chiselTests.TblSpec
```
I see errors like:
```
Starting High Form Check
Fatal Uncaught Exception.
NoFileInfo: [module Tbl]  Width cannot be negative or zero.
NoFileInfo: [module Tbl]  Width cannot be negative or zero.
TblTester5247337683591930572.fir@8.4: [module Tbl]  Width cannot be negative or zero.
   at /Users/chick/Aspire/IdeaProjects/firrtl/src/lib/stanza/compiler/lscore.lstanza@396.3
   at /Users/chick/Aspire/IdeaProjects/firrtl/src/lib/stanza/core/core.stanza@64.4
```

**Comment by ucbjrl** [comment]
retest this please

**Comment by chick** [comment]
Can this get merged now.  Adam's firrtl fix, fixed the errors

File: src/test/scala/chiselTests/TesterDriverSpec.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  val counter = Counter(1)
+
+  when(counter.inc()) {
+    stop()
+  }
+
+  val test_wire = Wire(UInt(1, width = test_wire_width))
+
+  // though we just set test_wire to 1, the assert below will pass because
+  // the finish will change it's value
```
Comment (ducky64) [inline src/test/scala/chiselTests/TesterDriverSpec.scala]: it's means "it is", I think you're looking for "its"

**Comment by ducky64** [comment]
One last grammar issue to fix.

**Comment by chick** [comment]
It's been changed from "it's" has been changed to "its"

**Comment by ducky64** [comment]
lgtm

**Pull request merged**